packages/nextjs/services/database/repositories/builds.ts

```diff
@@ -197,3 +198,28 @@ export const getBuildByBuildId = async (buildId: string) => {
   });
   return build;
 };
+
+export const getAllBuilds = async ({
+  category,
+  type,
+  nameSearch,
+}: {
+  category?: BuildCategory;
+  type?: BuildType;
+  nameSearch?: string;
+}) => {
+  const results = await db.query.builds.findMany({
+    where: and(
+      category ? eq(builds.buildCategory, category) : undefined,
+      type ? eq(builds.buildType, type) : undefined,
+      nameSearch ? ilike(builds.name, `%${nameSearch}%`) : undefined,
+    ),
+    with: {
+      likes: true,
+    },
+    orderBy: (builds, { desc, sql }) => [desc(sql`(SELECT COUNT(*) FROM build_likes WHERE build_id = ${builds.id})`)],
+    limit: 48,
```

Comment:
Instead of fetching all the builds we just limiting to top 48 builds, shall we have infinite scroll behavior on this page? Maybe we can tackle this in other PR? 

cc @carletex